modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -624,40 +648,56 @@ export default class ParticipantConnectionStatusHandler {
                 isInLastN} currentStatus => newStatus: ${
                 participant.getConnectionStatus()} => ${newState}`);
 
-        const nowMs = new Date().getTime();
-        const internalState = {
-            'instantMs': nowMs,
-            'isVideoMuted': isVideoMuted,
-            'isConnActiveByJvb': isConnActiveByJvb,
-            'isVideoTrackFrozen': isVideoTrackFrozen,
-            'inP2PMode': inP2PMode,
-            'inInLastN': isInLastN
-        };
-
-        if (!this.peerConnStatusMap[id] || this.peerConnStatusMap[id].state !== newState) {
-            // The peer connection status has changed. Compute the duration of the current
-            // connection status and send it as an analytics event.
-            if (this.peerConnStatusMap[id]) {
-                this.peerConnStatusMap[id].value = nowMs - this.peerConnStatusMap[id].startedMs;
-                Statistics.sendAnalytics(
-                    createParticipantConnectionStatusEvent(this.peerConnStatusMap[id]));
-            }
+        const oldConnectionStatus = this.connectionStatusMap[id] || {};
 
-            // And start a new status for the participant.
-            this.peerConnStatusMap[id] = {
-                'internalStates': [ internalState ],
-                'state': newState,
-                'startedMs': nowMs
-            };
-        } else {
-            // The connection status hasn't changed, but there was an internal state change.
-            // Register the internal state.
-            this.peerConnStatusMap[id].internalStates.push(internalState);
-        }
+        // Send an analytics event (guard on either the p2p flag or the connection status has changed
+        // since the last time this code block run).
+        if (!('p2p' in oldConnectionStatus)
+            || !('connectionStatus' in oldConnectionStatus)
+            || oldConnectionStatus.p2p !== inP2PMode
+            || oldConnectionStatus.connectionStatus !== newState) {
+
+            const nowMs = Date.now();
+
+            this.maybeSendParticipantConnectionStatusEvent(id, nowMs);
 
+            this.connectionStatusMap[id] = Object.assign(oldConnectionStatus, {
+                connectionStatus: newState,
+                p2p: inP2PMode,
+                startedMs: nowMs
+            });
+
+            // sometimes (always?) we're late to hook the TRACK_VIDEOTYPE_CHANGED event and the
+            // video type is not in oldConnectionStatus.
+            if (!('videoType' in this.connectionStatusMap[id])) {
+                this.connectionStatusMap[id].videoType = participant.getTracks().some(
```

Comment:
I'd select the first video track and return its videotype. Maybe you can use `JitsiParticipant.getTracksByMediaType`. I think it will be a little bit more clear what we try to do. WDYT?